pkg/util/gardener/util_test.go

```diff
@@ -6,12 +6,12 @@ import (
 )
 
 var _ = Describe("uti/gardener", func() {
-	Describe("IsShootCPNamespace", func() {
+	Describe("IsShootNamespace", func() {
 		It("should work expected on certain predefined values", func() {
-			Expect(IsShootCPNamespace("shoot--my-shoot")).To(BeTrue())
-			Expect(IsShootCPNamespace("shoot-my-shoot")).To(BeFalse())
-			Expect(IsShootCPNamespace("")).To(BeFalse())
-			Expect(IsShootCPNamespace("shoot--my--shoot")).To(BeTrue())
+			Expect(IsShootNamespace("shoot--my-shoot")).To(BeTrue())
```

Comment:
This should actually be false I think. Namespace could be either `shoot-<project-name>-<shoot-name>` (old format) or `shoot--<project-name>--<shoot-name>` (new format)

Reply:
IsShootNamespace() does not check as strictly as G/G. So I just added a comment in the code, explaining the discrepancy.

What I'm validating here is not that the name is correct, but that the second sequence of dash symbols does not interfere with interpreting the first one. It made more sense while I was only accepting '--': the test case was meant to verify that the function is using first-occurence-of-dash, and not last-occurence-of-dash.

Reply:
I'm marking this resolved, as we discussed if offline. Please reopen if necessary.